crates/autopilot/src/domain/competition/winner_selection/combinatorial.rs

```diff
@@ -350,3 +350,748 @@ type ScoreByDirection = HashMap<DirectedTokenPair, Score>;
+            )),
+            created: Default::default(),
+            valid_to: Default::default(),
+            signature: order::Signature::PreSign,
+            quote: None,
+        }
+    }
+
+    fn create_price(value: eth::U256) -> Price {
+        Price::try_new(eth::Ether(value)).unwrap()
+    }
+
+    fn create_auction(
+        orders: Vec<Order>,
+        prices: Option<HashMap<eth::TokenAddress, Price>>,
+    ) -> Auction {
+        // Initialize the prices of the tokens if they are not specified
+        let prices = prices.unwrap_or({
+            let default_price = create_price(DEFAULT_TOKEN_PRICE.into());
+            let mut res = HashMap::new();
+            for order in &orders {
+                res.insert(order.buy.token, default_price);
+                res.insert(order.sell.token, default_price);
+            }
+            res
+        });
+
+        Auction {
+            id: 0,
+            block: 0,
+            orders,
+            prices,
+            surplus_capturing_jit_order_owners: vec![],
+        }
+    }
+
+    fn create_trade(
+        order: &Order,
+        executed_sell: eth::U256,
+        executed_buy: eth::U256,
+    ) -> TradedOrder {
+        TradedOrder {
+            side: order::Side::Sell,
+            sell: order.sell,
+            buy: order.buy,
+            executed_sell: executed_sell.into(),
+            executed_buy: executed_buy.into(),
+        }
+    }
+
+    fn create_solution(
+        solution_id: u64,
+        solver_address: H160,
+        score: eth::U256,
+        trades: Vec<(OrderUid, TradedOrder)>,
+        prices: Option<HashMap<TokenAddress, Price>>,
+    ) -> Participant<Unranked> {
+        // The prices of the tokens do not affect the result but they keys must exist
+        // for every token of every trade
+        let prices = prices.unwrap_or({
+            let mut res = HashMap::new();
+            for (_, trade) in &trades {
+                res.insert(trade.buy.token, create_price(eth::U256::one()));
+                res.insert(trade.sell.token, create_price(eth::U256::one()));
+            }
+            res
+        });
+
+        let trade_order_map: HashMap<OrderUid, TradedOrder> = trades.into_iter().collect();
+        let solver_address = eth::Address(solver_address);
+
+        let solution = Solution::new(
+            solution_id,
+            solver_address,
+            Score(eth::Ether(score)),
+            trade_order_map,
+            prices,
+        );
+
+        Participant::new(
+            solution,
+            Driver::mock(solver_address.to_string(), solver_address).into(),
+        )
+    }
+
+    fn amount(value: u128) -> String {
+        // adding decimal units to avoid the math rounding it down to 0
+        (value * 10u128.pow(15)).to_string()
+    }
+
+    fn score(score: u128) -> String {
+        let score: u128 = amount(score)
+            .parse()
+            .expect("Failed to parse score as u128");
```

Comment:
Why do you pass in a `u128` and then parse it instead of just using the value as is?

Reply:
Leftover from previous version of the code. Fixed now.